src/librustc_typeck/astconv.rs

```diff
@@ -1398,50 +1401,53 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             if trait_ref.self_ty() != dummy_self {
-                // FIXME: There appears to be a missing filter on top of `expand_trait_aliases`,
-                // which picks up non-supertraits where clauses - but also, the object safety
-                // completely ignores trait aliases, which could be object safety hazards. We
-                // `delay_span_bug` here to avoid an ICE in stable even when the feature is
-                // disabled. (#66420)
-                tcx.sess.delay_span_bug(DUMMY_SP, &format!(
-                    "trait_ref_to_existential called on {:?} with non-dummy Self",
-                    trait_ref,
-                ));
+                bug!("trait_ref_to_existential called on {:?} with non-dummy Self", trait_ref);
             }
             ty::ExistentialTraitRef::erase_self_ty(tcx, trait_ref)
         };
 
         // Erase the `dummy_self` (`trait_object_dummy_self`) used above.
-        let existential_trait_refs = regular_traits.iter().map(|i| {
-            i.trait_ref().map_bound(|trait_ref| trait_ref_to_existential(trait_ref))
-        });
-        let existential_projections = bounds.projection_bounds.iter().map(|(bound, _)| {
-            bound.map_bound(|b| {
-                let trait_ref = trait_ref_to_existential(b.projection_ty.trait_ref(tcx));
-                ty::ExistentialProjection {
-                    ty: b.ty,
-                    item_def_id: b.projection_ty.item_def_id,
-                    substs: trait_ref.substs,
-                }
-            })
-        });
+        let existential_trait_refs = regular_traits
+            .iter()
+            .map(|i| i.map_bound(|trait_ref| trait_ref_to_existential(trait_ref)));
+        let existential_projections = bounds.projection_bounds
+            .iter()
+            .map(|(bound, _)| {
+                bound.map_bound(|b| {
+                    let trait_ref = trait_ref_to_existential(b.projection_ty.trait_ref(tcx));
+                    ty::ExistentialProjection {
+                        ty: b.ty,
+                        item_def_id: b.projection_ty.item_def_id,
+                        substs: trait_ref.substs,
+                    }
+                })
+            });
 
         // Calling `skip_binder` is okay because the predicates are re-bound.
-        let regular_trait_predicates = existential_trait_refs.map(
-            |trait_ref| ty::ExistentialPredicate::Trait(*trait_ref.skip_binder()));
-        let auto_trait_predicates = auto_traits.into_iter().map(
-            |trait_ref| ty::ExistentialPredicate::AutoTrait(trait_ref.trait_ref().def_id()));
+        let regular_trait_predicates = existential_trait_refs
+            .map(|trait_ref| ty::ExistentialPredicate::Trait(*trait_ref.skip_binder()));
+        let auto_trait_predicates = auto_traits
+            .into_iter()
+            .map(|trait_ref| ty::ExistentialPredicate::AutoTrait(trait_ref.def_id()));
```

Comment:
To confirm, also no functional changes here, correct? ( I didn't spot any )

Reply:
Correct, just made it more readable since I was editing this fn anyway.